src/future.ml

```diff
@@ -503,6 +622,9 @@ module Scheduler = struct
 
   let go ?(log=Log.no_log) t =
     Lazy.force Ansi_color.setup_env_for_opam_colors;
+    Log.info log ("Workspace root: " ^ !Clflags.workspace_root);
+    if !Clflags.workspace_root <> Filename.current_dir_name then
+      Sys.chdir !Clflags.workspace_root;
```

Comment:
The chdir need to happen before `Main.setup`, it could probably stay in `set_common` for now